grafana/resource_datasource_permission.go

```diff
@@ -118,16 +128,22 @@ func ReadDatasourcePermissions(ctx context.Context, d *schema.ResourceData, meta
 		return diag.FromErr(err)
 	}
 
-	permissionItems := make([]interface{}, len(response.Permissions))
-	for i, permission := range response.Permissions {
+	var permissionItems []interface{}
```

Comment:
NIT: You should use the make function to construct the slice because we know the required size. But instead of setting the size as it was before we can set the capacity and still use the append function `make([]interface{}, 0, len(response.Permissions))`